src/main/java/net/neoforged/neoforge/common/data/internal/NeoForgeDamageTypeTagsProvider.java

```diff
@@ -92,6 +118,22 @@ protected void addTags(HolderLookup.Provider lookupProvider) {
         tagWithOptionalLegacy(Tags.DamageTypes.NO_FLINCH);
     }
 
+    private TagAppender<DamageType> addAsVanilla(ResourceKey<DamageType> entry) {
+        final List<TagBuilder> builders = new ArrayList<>();
+        vanillaBuilders.forEach((location, tagBuilder) -> {
+            if (tagBuilder.build().stream().anyMatch(tagEntry -> tagEntry.verifyIfPresent(element -> element.equals(entry.location()), tag -> false))) {
```

Comment:
I don't particularly like that this means the tags are built to check, but I don't think that outweighs adding an accessor to the `TagBuilder#entries` field. (Just a note.)